spec/datadog/tracing/tracer_spec.rb

```diff
@@ -259,29 +273,55 @@
       end
 
       context 'when nesting spans' do
-        it 'propagates parent span and uses default service name' do
+        subject!(:nested_spans) do
```

Comment:
Can we remove `!` here?

This brings confusion to me with the ordering of execution, such as "Would this be executed before the `let` in the nested context"?